### PR TITLE
run: Add --uts flag

### PR DESCRIPTION
**Pull request by dcantah**
Adds support for specifying the host UTS namespace (the only thing
docker supported). This also consolidates the namespace options
checks for linux on run into a single function.

### CODE REVIEW

**Comment by dcantah** [comment]
Unfortunately don't have a decent box to verify this for a day or two, so draft for now

**Comment by dcantah** [comment]
Same strange (potential) rate-limit behavior on the project checks. If anyone can re-kick that off I'd appreciate

**Comment by Zheaoli** [comment]
Thanks for your PR! In my personal opinion, I think it's unnecessary to change the `setPlatformOptions` signature in this PR. One PR for one thing, please.

If we need to change all signatures related to `internalLabels`, maybe we need another issue to discuss it.

**Comment by dcantah** [comment]
It's not worth the hassle then don't worry about it, I'll move the signature change to another PR then. It was impacting the ability to just easily setup all the pid options enough it seemed warranted, and it's not exported or some relied on API here.

**Comment by dcantah** [comment]
@Zheaoli, removed the setPlatformOptions signature change. I'll make an issue/separate PR

**Comment by AkihiroSuda** [comment]
> @Zheaoli, removed the setPlatformOptions signature change. I'll make an issue/separate PR. Thanks!

Doesn't seem removed

**Comment by AkihiroSuda** [comment]
A separate PR is preferred, but a separate commit in the same PR is also ok IMHO

**Comment by dcantah** [comment]
> > @Zheaoli, removed the setPlatformOptions signature change. I'll make an issue/separate PR. Thanks!
> 
> Doesn't seem removed

It's the same signature still, it's just over multiple lines as it was a bit long. The signature change was:

### Before 
```go
func setPlatformOptions(
 	ctx context.Context,
 	opts []oci.SpecOpts,
 	cmd *cobra.Command,
 	client *containerd.Client,
 	id string,
 	internalLabels internalLabels,
 ) ([]oci.SpecOpts, internalLabels, error) {
```

### After 
```go
func setPlatformOptions(
 	ctx context.Context,
 	opts []oci.SpecOpts,
 	cmd *cobra.Command,
 	client *containerd.Client,
 	id string,
 	internalLabels *internalLabels,
 ) ([]oci.SpecOpts, error) {
```

but I reverted the `after` variant after the discussion

**Comment by dcantah** [comment]
> A separate PR is preferred, but a separate commit in the same PR is also ok IMHO (for this amount of PR)

Yea, this didn't seem too unwieldy to all be in one PR, but I'll just make a separate at this point no worries.

**Comment by dcantah** [comment]
Rebased after https://github.com/containerd/nerdctl/pull/1771 landed.

File: cmd/nerdctl/run_linux.go [hunk 1]
```diff
@@ -136,17 +136,6 @@ func setPlatformOptions(
...
-	if err != nil {
-		return nil, err
-	}
-	pidOpts, pidInternalLabel, err := generatePIDOpts(ctx, client, pid)
-	if err != nil {
-		return nil, err
-	}
-	internalLabels.pidContainer = pidInternalLabel
-	opts = append(opts, pidOpts...)
-
```
Comment (AkihiroSuda) [inline cmd/nerdctl/run_linux.go:149]: The pidns changes seem irrelevant to utsns, could you consider splitting the commit? Not necessary to open a separate PR.

Comment (dcantah) [reply cmd/nerdctl/run_linux.go:149]: Sure, will try and get to today

Comment (dcantah) [reply cmd/nerdctl/run_linux.go:149]: @AkihiroSuda done

**Review by AkihiroSuda — APPROVED** [review]
Thanks
